### PR TITLE
WTCET-35

### PR DESCRIPTION
**Pull request by PaddyCello**
Hey Puneet! Here's the first PR for the Bank Account project - please let me know your thoughts.

Just as a heads up, I've created a method (Admin, from line 85) for validating the data from the CSV file; but as the CSV provided from the Udemy course has all of the data in the correct format, I haven't really been able to test it. However, I could attempt to make a dummy CSV file for testing purposes, with incorrect data types in certain places etc; if this would be worthwhile and not an unnecessary distraction, please let me know and I'll give it a shot. :)

### CODE REVIEW

File: src/main/java/com/pjohnson_wtc/bank_account/Admin.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+			logger.log(Level.WARNING, "Cannot open an account without funds");
+			return false;
+		}
+		
+		//If all is good, return true
+		return true;
+	}
+	
+	//Method for creating a single account
+	private Account finallyCreateAccount(List<String> applicant) {
```
Comment (puneetg1) [inline src/main/java/com/pjohnson_wtc/bank_account/Admin.java]: CreateAccount will make better sense

File: src/main/java/com/pjohnson_wtc/bank_account/Admin.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+			logger.log(Level.WARNING, "Could not read file", ioe.toString());
+			return null;
+		}
+	}
+	
+	//Method for validating input from csv
+	private boolean validateApplicant(List<String> applicant) {
+		
+		//Return false if account type is neither Savings nor Checking
+		if (!(applicant.get(2).equals("Savings") || applicant.get(2).equals("Checking"))) {
```
Comment (puneetg1) [inline src/main/java/com/pjohnson_wtc/bank_account/Admin.java]: Null check missing. It's a run time exception and should be handled

File: src/main/java/com/pjohnson_wtc/bank_account/Admin.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+			
+			logger.log(Level.WARNING, "Valid account type required in column 3");
+			return false;
+		}
+		
+		//Return false if parsing numeric values from SSN or balance throws an Exception
+		try {
```
Comment (puneetg1) [inline src/main/java/com/pjohnson_wtc/bank_account/Admin.java:87]: put everything inside one null

**Review by puneetg1 — CHANGES_REQUESTED** [review]

File: src/main/java/com/pjohnson_wtc/bank_account/Admin.java (hunk 1 above)
Comment (PaddyCello) [reply src/main/java/com/pjohnson_wtc/bank_account/Admin.java]: Very true - that's probably also a sign that I should change the name of another method I've given that name to 😂

File: src/main/java/com/pjohnson_wtc/bank_account/Admin.java (hunk 2 above)
Comment (PaddyCello) [reply src/main/java/com/pjohnson_wtc/bank_account/Admin.java]: Good point - will do

File: src/main/java/com/pjohnson_wtc/bank_account/Admin.java (hunk 3 above)
Comment (PaddyCello) [reply src/main/java/com/pjohnson_wtc/bank_account/Admin.java:87]: Fab, will do

**Comment by PaddyCello** [comment]
Right, that's the changes made; additionally, I renamed the method that starts on Admin, Line 22 to "createAccountsFromCsv()". Let me know if all is good!

**Review by puneetg1 — APPROVED** [review]

**Comment by puneetg1** [comment]
Perfect Thanks !!!

**Pull request merged**